device/include/ScalarRegisterAccessor.h

```diff
@@ -104,6 +104,18 @@ namespace ChimeraTK {
      */
     void writeIfDifferent(UserType newValue, VersionNumber versionNumber = VersionNumber{nullptr});
 
+    /**
+     * Convenience function to set and write new value. The given version number
+     * is only used in case the value differs. If versionNumber == {nullptr}, a new version number is generated only if
```

Comment:
The comment is wrong (copy and paste error from writeIfDifferent). The value is always written, not only when it's different.
Hence, the whole scheme with nullprt is not needed.